src/Compilers/CSharp/Portable/Lowering/LocalRewriter/DelegateCacheRewriter.cs

```diff
@@ -105,23 +105,23 @@ private DelegateCacheContainer GetOrAddCacheContainer(BoundDelegateCreationExpre
         }
         else
         {
-            var containers = _genericCacheContainers ??= new Dictionary<MethodSymbol, DelegateCacheContainer>(ReferenceEqualityComparer.Instance);
+            var containers = _genericCacheContainers ??= new Dictionary<Symbol, DelegateCacheContainer>(ReferenceEqualityComparer.Instance);
 
-            if (containers.TryGetValue(ownerFunction, out container))
+            if (containers.TryGetValue(owner, out container))
             {
                 return container;
             }
 
-            container = new DelegateCacheContainer(ownerFunction, _topLevelMethodOrdinal, containers.Count, generation);
-            containers.Add(ownerFunction, container);
+            container = new DelegateCacheContainer(_factory.CompilationState.Type, owner, _topLevelMethodOrdinal, ownerUniqueId: containers.Count, generation);
+            containers.Add(owner, container);
         }
 
         _factory.AddNestedType(container);
 
         return container;
     }
 
-    private static bool TryGetOwnerFunction(MethodSymbol currentFunction, BoundDelegateCreationExpression boundDelegateCreation, [NotNullWhen(true)] out MethodSymbol? ownerFunction)
+    private static bool TryGetOwnerFunction(MethodSymbol currentFunction, BoundDelegateCreationExpression boundDelegateCreation, [NotNullWhen(true)] out Symbol? owner)
```

Comment:
>TryGetOwnerFunction [](http://example.com/codeflow?start=24&length=19)

The name should probably be updated since the owner may not be a function
nit: consider adding an example with non-function owner int he comment

Reply:
> The name should probably be updated since the owner may not be a function

I will rename in one of the next PRs